src/Analyzers/CSharp/Tests/RemoveRedundantElseStatement/RemoveRedundantElseStatementTests.cs

```diff
@@ -0,0 +1,301 @@
+                    }
+                }
+                """, """
+                using System;
+                
+                class C
+                {
+                    int Count(int n) 
+                    {
+                        int i = 0;
+                        while (true)
+                        {
+                            if (i == n)
+                            {
+                                break;
+                            }
+
+                            i++;
+                        }
+                
+                        return i;
+                    }
+                }
+                """);
+        }
+
+        [Fact]
+        public async Task TestRedundantElseFix_IfElseWithContinue()
+        {
+            await VerifyCS.VerifyCodeFixAsync("""
+                using System;
+
+                class C
+                {
+                    int Count(int n) 
+                    {
+                        int i = 0;
+                        while (true)
+                        {
+                            if (i < n)
+                            {
+                                i+=1;
+                                continue;
+                            }
+                            [|else|]
+                            {
+                                return i;
+                            }
+                        }
+                    }
+                }
+                """, """
+                using System;
+                
+                class C
+                {
+                    int Count(int n) 
+                    {
+                        int i = 0;
+                        while (true)
+                        {
+                            if (i < n)
+                            {
+                                i+=1;
+                                continue;
+                            }
+
+                            return i;
+                        }
+                    }
+                }
+                """);
+        }
+
+        [Fact]
+        public async Task TestRedundantElseFix_IfElseWithoutContinue()
+        {
+            await new VerifyCS.Test
+            {
+                TestCode = """
+                    using System;
+                    
+                    class C
+                    {
+                        int Count(int n) 
+                        {
+                            int i = 0;
+                            while (true)
+                            {
+                                if (i < n)
+                                {
+                                    i+=1;
+                                }
+                                else
+                                {
+                                    return i;
+                                }
+                            }
+                        }
+                    }
+                    """
+            }.RunAsync();
```

Comment:
tests for global statements and switch cases as well plz.
tests with semantic issues caused by not having the dedicated 'else' scope.
test that shows that the code in the else-block is formatted poorly, but all that the fix does is dedent thing
tests that show that fix-all works.  i.e. an if/else inside of an if/else.

Reply:
Again I'm having issues getting tests to pass for global statements...
![image](https://github.com/dotnet/roslyn/assets/46457277/7fbb7038-99a8-41ce-9879-94e74dcc591e)

When I remove that line I get different CS8805 error I can't seem to get around

Reply:
Thanks to @Youssef1313 I was able to go around CS8805 error, however this newline is still missing, is there a way to add this manually? I tried adding trailing EndOfLineTrivia to the global statement but it didn't work

Reply:
@CyrusNajmabadi any updates on this?

Reply:
is it possible you have elastic trivia on one of those nodes/tokens?   if so, the formatting engine is probably deleting this line.  try removing all the trivia once you make the node and put it the trivia you want explicitly.